ksql-serde/src/test/java/io/confluent/ksql/serde/kafka/KafkaSerdeFactoryTest.java

```diff
@@ -192,7 +194,11 @@ private void shouldHandle(final SqlType fieldSchema, final Object value) {
     final Object result = serde.deserializer().deserialize("topic", bytes);
 
     // Then:
-    assertThat(result, is(struct));
+    if (value == null) {
+      assertThat(result, is(nullValue()));
+    } else {
+      assertThat(result, is(struct));
+    }
```

Comment:
if its null, then assert its null... weird asserting! ;)

Also, this means all of the tests will pass if deserialize always returns `null`.

I think it would be more explicit if we revert this change and instead have explicit tests that test we:

a) serialize `null` as `null` , and
b) deserialize `null` as `null`.

Just played around with this, so may as well share the code.

i.e. we revert this to:

```java
  assertThat(result, is(struct));
```

And replace `shouldHandleNulls` with:

```java
@Test
  public void shouldSerializeNullAsNull() {
    // Given:
    final PersistenceSchema schema = schemaWithFieldOfType(SqlTypes.INTEGER);

    final Serde<Object> serde = factory.createSerde(schema, ksqlConfig, srClientFactory);

    // When:
    final byte[] result = serde.serializer().serialize("topic", null);

    // Then:
    assertThat(result, is(nullValue()));
  }

  @Test
  public void shouldDeserializeNullAsNull() {
    // Given:
    final PersistenceSchema schema = schemaWithFieldOfType(SqlTypes.INTEGER);

    final Serde<Object> serde = factory.createSerde(schema, ksqlConfig, srClientFactory);

    // When:
    final Object result = serde.deserializer().deserialize("topic", null);

    // Then:
    assertThat(result, is(nullValue()));
  }
```